test/cljam/t_bam.clj

```diff
@@ -40,23 +45,96 @@
 ;;     (is (not-throw? (spit-bam-for-test temp-file
 ;;                                        (slurp-bam-for-test large-bam-file))))))
 
+(defn- shallow= [alns1 alns2]
+  (when (= (count alns1) (count alns2))
+    (loop [alns1 alns1
+           alns2 alns2]
+      (let [aln1 (first alns1)
+            aln2 (first alns2)]
+        (if (nil? aln1)
+          true
+          (when (and (= (:rname aln1) (:rname aln2))
+                     (= (:pos aln1) (:pos aln2)))
+            (recur (rest alns1) (rest alns2))))))))
+
+(defn- pointer= [alns1 alns2]
+  (when (= (count alns1) (count alns2))
+    (loop [alns1 alns1
+           alns2 alns2]
+      (let [aln1 (first alns1)
+            aln2 (first alns2)]
+        (if (nil? aln1)
+          true
+          (when (and (= (:rname aln1) (:rname aln2))
+                     (= (:pos aln1) (:pos aln2))
+                     (= (:flag aln1) (:flag aln2)))
+            (recur (rest alns1) (rest alns2))))))))
+
 (deftest bamreader
   (with-before-after {:before (do (prepare-cache!)
                                   (spit-bam-for-test temp-file test-sam))
                       :after (clean-cache!)}
-    (let [rdr (bam/reader temp-file :ignore-index true)]
-      (is (= (io/read-refs rdr) test-sam-refs)))))
+    (let [rdr (bam/reader temp-file :ignore-index false)]
```

Comment:
BAMReader will not be closed. `with-open` should be used instead of `let`. (I don't know why original code uses `let`...)